north/src/runtime/state.rs

```diff
@@ -332,19 +367,44 @@ impl State {
         }
 
         for (_name, container) in self.applications.values().map(|a| (a.name(), &a.container)) {
-            umount_npk(container).await.map_err(Error::Mount)?;
+            let wait_for_dm = self
+                .repositories
+                .get(&container.repository)
+                .map(|r| r.key.is_some())
+                .unwrap_or(false);
```

Comment:
Why is it relevant if a key exists to know if we want to wait for the device mapping?

Reply:
yes, this is pretty much a hack. Checking for the key here indicates whether the container is mounted through verity or directly in the loopback device. In the verity case, we have to wait for the device mapper device to  be removed indirectly after removing the loop device.

Reply:
ok..then I'd call it rather something like `verity_dm_used`